controllers/remote/cluster_cache_reconciler.go

```diff
@@ -32,6 +32,8 @@ import (
 
 // ClusterCacheReconciler is responsible for stopping remote cluster caches when
 // the cluster for the remote cache is being deleted.
+//
+// Deprecated: This will be removed in Cluster API v1.10, use clustercache.ClusterCache instead.
```

Comment:
I think it makes sense to give folks on release time to migrate away (ClusterCacheTracker would be removed ahead of the v1.10 (April 2025) release).

Folks can always just copy the old ClusterCacheTracker if they would continue to use it.

Reply:
works for me

Reply:
To what do you want to change it?

(Just adding the deprecation itself is not a breaking change)

Reply:
Added a note here: https://github.com/kubernetes-sigs/cluster-api/issues/11092#issuecomment-2426337610